File: app/src/main/java/harkor/mycryptocurrency/Details.java
package harkor.mycryptocurrency;

import android.os.Parcel;
import android.os.Parcelable;

public class Details implements Parcelable {
    public final String name;
    public final String date;

    public Details(String name, String date) {
        this.name = name;
        this.date = date;
    }


    protected Details(Parcel in) {
        name = in.readString();
        date = in.readString();
    }

    public static final Creator<Details> CREATOR = new Creator<Details>() {
        @Override
        public Details createFromParcel(Parcel in) {
            return new Details(in);
        }

        @Override
        public Details[] newArray(int size) {
            return new Details[size];
        }
    };

    @Override
    public int describeContents() {
        return 0;
    }

    @Override
    public void writeToParcel(Parcel parcel, int i) {
        parcel.writeString(name);
    }
}

File: app/src/main/java/harkor/mycryptocurrency/Details.kt
package harkor.mycryptocurrency

import android.os.Parcel
import android.os.Parcelable

class Details : Parcelable {
    val name: String?
    val date: String?

    constructor(name: String, date: String) {
        this.name = name
        this.date = date
    }


    protected constructor(`in`: Parcel) {
        name = `in`.readString()
        date = `in`.readString()
    }

    override fun describeContents(): Int {
        return 0
    }

    override fun writeToParcel(parcel: Parcel, i: Int) {
        parcel.writeString(name)
    }

    companion object {

        val CREATOR: Parcelable.Creator<Details> = object : Parcelable.Creator<Details> {
            override fun createFromParcel(`in`: Parcel): Details {
                return Details(`in`)
            }

            override fun newArray(size: Int): Array<Details?> {
                return arrayOfNulls(size)
            }
        }
    }
}
